### ISSUE TITLE
Any plan to upgrade spring-boot-3?

### ISSUE THREAD

**Issue by evijayan2** [adobe/S3Mock#1008]
Hi Team,

Any plan to upgrade spring-boot-3?

**Comment by afranken** [comment]
Yes, I have a branch ready:
https://github.com/adobe/S3Mock/pull/833

The problem is that this update breaks all compatibility with older releases.

In general, it is discouraged to use S3Mock directly as a Java dependency in tests:
https://github.com/adobe/S3Mock#s3mock-java

Please think about migrating your tests to use Docker:
https://github.com/adobe/S3Mock#s3mock-docker

**Comment by evijayan2** [comment]
Thanks, we will plan that.

**Comment by stefan-prange** [comment]
Is there any chance to have the version 3.0.0-alpha be released to Maven central?
The Spring Boot 3 migration of our applications is blocked by the lacking Spring-Boot-3-compatibility of S3Mock.
Unfortunately, using S3Mock from a Docker container is not an option in our testing set-up.

**Comment by stefan-prange** [comment]
Ok, no need to hurry. I managed to migrate our test cases from S3Mock to Mockito and a self-built AmazonS3 implementation.

**Comment by thomaschilton-viavi** [comment]
@stefan-prange can you share more details of your Mockito implementation?  I'm in a similar position where the testcontainers approach is not viable due to using a Docker build CI infrastructure - and cannot move other libraries forward due to the s3mock dependencies.

**Comment by stefan-prange** [comment]
@thomaschilton-viavi
well, it's quite straight forward. This is a Spring configuration class that I use in my test case and that creates an AmazonS3 client with mocked behavior. Luckily, my use case was only about reading a single file from S3.
```java
@Configuration
public class UnittestS3ClientFactory {

    AmazonS3 mockAmazonS3;

    @Bean
    @Primary // primary in order to override the regular AmazonS3 bean creation.
    @Qualifier("mySpecialPurposeS3Client")
    public AmazonS3 produceS3MockS3Client() {

        mockAmazonS3 = Mockito.mock(AmazonS3.class);

        String testBucketName = "my-bucket", s3Path = "myTestFile.txt";
        S3Object s3Object;
        try {
            s3Object = buildS3Object();
            when(mockAmazonS3.getObject(new GetObjectRequest(testBucketName, s3Path))).thenReturn(s3Object);
        } catch (FileNotFoundException e) {
            throw new RuntimeException("Failed to initialize mock S3Object: " + e.getMessage());
        }
        return mockAmazonS3;
    }

    private S3Object buildS3Object() throws FileNotFoundException {
        S3Object s3Object = new S3Object();
        s3Object.setObjectContent(new FileInputStream("./src/test/resources/test-file.txt"));
        ObjectMetadata metaData = new ObjectMetadata();
        metaData.setLastModified(new Date());
        s3Object.setObjectMetadata(metaData);
        return s3Object;
    }
}
```

**Comment by afranken** [comment]
@stefan-prange / @thomaschilton-viavi / @evijayan2 
I just released S3Mock 3.0.0 which updates Spring Boot to 3.1.0 / Spring Framework 6.0.9

Still, I would like to urge you to try and migrate your test setup to use Docker - otherwise, you'll always depend on S3Mock using the same Spring versions as your app...

**Comment by stefan-prange** [comment]
@afranken 
Thanks a lot. I followed your advice to remove the S3Mock dependency, but in a different way:
I replaced S3Mock everywhere with either a Mockito based AmazonS3 mock or with subclasses of AbstractAmazonS3. The latter only serve my testing needs and are otherwise left empty (i.e, most of its methods are stubs that throw an UnsupportedOperationException). I'm quite happy with that solution.
